src/loader.rs

```diff
@@ -25,28 +53,196 @@ impl AssetLoader for ObjLoader {
     }
 }
 
+#[cfg(feature = "scene")]
+impl FromWorld for ObjLoader {
+    fn from_world(world: &mut World) -> Self {
+        let supported_compressed_formats = match world.get_resource::<RenderDevice>() {
+            Some(render_device) => CompressedImageFormats::from_features(render_device.features()),
+
+            None => CompressedImageFormats::all(),
+        };
+        Self {
+            supported_compressed_formats,
+        }
+    }
+}
+
 #[derive(Error, Debug)]
 pub enum ObjError {
     #[error("Invalid OBJ file: {0}")]
-    InvalidFile(#[from] tobj::LoadError),
+    TobjError(#[from] tobj::LoadError),
+    #[cfg(feature = "scene")]
+    #[error("Invalid image file for texture: {0}")]
+    InvalidImageFile(PathBuf),
+    #[error("Asset reading failed: {0}")]
+    AssetIOError(#[from] bevy_asset::AssetIoError),
+    #[error("Texture conversion failed: {0}")]
+    TextureError(#[from] bevy_render::texture::TextureError),
 }
 
 async fn load_obj<'a, 'b>(
     bytes: &'a [u8],
     load_context: &'a mut LoadContext<'b>,
+    #[cfg(feature = "scene")] supported_compressed_formats: CompressedImageFormats,
 ) -> Result<(), ObjError> {
-    let mesh = load_obj_from_bytes(bytes)?;
-    load_context.set_default_asset(LoadedAsset::new(mesh));
+    #[cfg(not(feature = "scene"))]
+    let obj = load_obj_from_bytes(bytes)?;
+    #[cfg(feature = "scene")]
+    let obj = load_obj_from_bytes(bytes, load_context, supported_compressed_formats).await?;
+
+    load_context.set_default_asset(LoadedAsset::new(obj));
     Ok(())
 }
 
-fn load_mtl(_path: &std::path::Path) -> tobj::MTLLoadResult {
-    Err(tobj::LoadError::OpenFileFailed)
+#[cfg(feature = "scene")]
+async fn load_texture_image<'a, 'b>(
+    image_path: &'a str,
+    load_context: &'a mut LoadContext<'b>,
+    supported_compressed_formats: CompressedImageFormats,
+) -> Result<(Image, String), ObjError> {
+    let path = Path::new(&image_path);
+    let filename = path
+        .to_str()
+        .ok_or(ObjError::InvalidImageFile(path.to_path_buf()))?;
+    let extension = ImageType::Extension(
+        path.extension()
+            .and_then(|e| e.to_str())
+            .ok_or(ObjError::InvalidImageFile(path.to_path_buf()))?,
+    );
+    let bytes = load_context.asset_io().load_path(path).await?;
+    // TODO(luca) confirm value of is_srgb
+    let is_srgb = true;
+    Ok((
+        Image::from_buffer(&bytes, extension, supported_compressed_formats, is_srgb)?,
+        filename.to_string(),
+    ))
+}
+
+#[cfg(feature = "scene")]
+async fn load_obj_from_bytes<'a, 'b>(
+    mut bytes: &'a [u8],
+    load_context: &'a mut LoadContext<'b>,
+    supported_compressed_formats: CompressedImageFormats,
+) -> Result<Scene, ObjError> {
+    let options = tobj::GPU_LOAD_OPTIONS;
+    let asset_io = &load_context.asset_io();
+    let obj = tobj::load_obj_buf_async(&mut bytes, &options, |p| async move {
+        asset_io
```

Comment:
I'm not sure about this, I see other asset loading crates (i.e. `bevy_gltf`) use `bevy_log` instead of log, do you think it could be possible, or even a good idea, to use that instead?

Reply:
Of course you can use that! It's probably a good idea, because then library users can control the logging from the Log Plugin like this:

https://bevy-cheatbook.github.io/features/log.html

Reply:
With https://github.com/AmionSky/bevy_obj/pull/12/commits/74bba9b3a1616421ed2237df57196ab15f2728cf we don't use material names anymore so the specific issue showed in your log shouldn't happen anymore, unless `tobj` fails and returns a material index that points past the end of the material vector, which is hopefully unlikely enough.

I'm quite on the fence with this now and will defer to @AmionSky on whether we should add logging and how verbose it should be, since the original implementation has no logging at all

Reply:
Personally I would say we don't need this level of logging. The tobj logs already say which mesh and material was the issue. But also we should error out on a missing material with a better error message (maybe also saying which obj file caused the issue (or better error messages in general))